error404.lp

```diff
@@ -0,0 +1,26 @@
+<? --[[
+*  Pi-hole: A black hole for Internet advertisements
+*  (c) 2023 Pi-hole, LLC (https://pi-hole.net)
+*  Network-wide ad blocking via your own hardware.
+*
+*  This file is copyright under the latest version of the EUPL.
+*  Please see LICENSE file for your rights under this license.
+--]]
+
+mg.include('scripts/pi-hole/lua/header.lp','r')
+?>
+<body class="hold-transition layout-boxed login-page">
+    <div class="box login-box">
+        <section style="padding: 15px;">
+                <h2 class="error-headline text-yellow">404</h2>
+                <div class="error-content">
+                    <h3><i class="fa fa-warning text-yellow"></i> Oops! Page not found.</h3>
+                    <p>
+                        We could not find the page you were looking for.<br>
+                        Meanwhile, you may want toreturn to <a href="/">the dashboard</a> .
```

Comment:
The link is pointing to the root.

```suggestion
                        Meanwhile, you may want to return to <a href="/admin">the dashboard</a> .
```

Reply:
Is this an issue? We have an automatic redirect...?

Reply:
This is what I see:

![404_root](https://github.com/pi-hole/web/assets/1385443/ed42ae02-c70b-4113-8f01-a9c5e4fc2ff4)

Reply:
You are correct, it's broken when accessing the web interface by IP. However, it works when accessing via hostname.

Reply:
The code linked in my comment above will fix it. Basically, change the / to <?=pihole.webhome()?>

Reply:
Apparently the 404 page is not used with IP.

The address `http://192.168.0.206/admin/invalidpage` doesn't show the 404 page. It shows the text error message:
```
Error 404: Not Found
Not found
```

Reply:
Are you using the corresponding updated FTL code also proposed by @DL6ER? It won't show error pages without it.

Reply:
After restarting the container, the 404 page is working, but the issue with IP remains.

Reply:
I'm using IP and the  ?=pihole.webhome()? link works. 
![image](https://github.com/pi-hole/web/assets/119230128/ac58f8ac-1b1c-4c5f-a655-d895fae9fb2e)

Reply:
Is the issue with IP because if you use the IP address, e.g., 192.168.1.44/ you get a 403 directory error instead of a 404 error, and the error_pages only works for 404 pages as presently coded? 192.168.1.44/oops gives a 404 for example. Changes to make pages for 403 and 500 errors, etc could be achieved.

Reply:
Not sure how, but I missed this entire conversation. Should be fixed with the latest commit.